### PR TITLE
Replace supported image extension list with image detection via MagickImageInfo.

### LINKED ISSUE #802: Add support for .dds images?
Hello. I use programs that require me to make use of .dds images and this program does not preview the file correctly and neither does any other plugins for this program. It would be more convenient for me to have this program preview .dds files so I can look at those images more quickly.

Please consider adding support for this file format.

### PR DESCRIPTION
**Pull request by mooflu**
Fixes #802 #768

### CODE REVIEW

**Comment by xupefei** [comment]
My main concern is ImageMagick handles way more file types than images, such as PDF.
Could you adjust the Property of this plugin to -4 (1 higher than TextViewer) so that it will not steal preview of other plugins?

**Comment by mooflu** [comment]
Good point. With ffmpeg installed it'll also try to handle video formats. VideoViewer has priority -10. I guess prio values would need to be VideoViewer > ImageViewer > TextViewer. E.g. -3, -4, -5 ; comment in VideoViewer  says "make it lower than TextViewer" but not why. Will it detect some video files as text in TextViewer?

**Comment by mooflu** [comment]
It was changed to lower than TextViewer here: https://github.com/QL-Win/QuickLook/commit/24c03163f68de7589f03d4e23a493cea9ce0a87c

I see .ts as a video file extension, so if video was > text it would try to view typescript files as video. So I can see why video would need to be lower that text if video only goes by file extension. :(

**Comment by mooflu** [comment]
Tried to do the same in VideoViewer and detect audio/video file via MediaInfo instead of file extensions. See latest WIP - currently re-creates MediaInfo in both CanHandle and Prepare. Out of time for today...

**Comment by mooflu** [comment]
Build artifact for the latest here:https://ci.appveyor.com/project/mooflu/quicklook-moar-image-types/build/artifacts

**Comment by Michael19723** [comment]
Does it also support .jfif files?

**Comment by mooflu** [comment]
@Michael82548 yes, should work. Both in the nightly build (https://github.com/QL-Win/QuickLook#downloadinstallation) as well as the test build for this merge request above.

**Comment by xupefei** [comment]
I'm quite busy in the past few days. Will try to review this PR by this weekend.

File: QuickLook.Plugin/QuickLook.Plugin.ImageViewer/Plugin.cs [hunk 1]
```diff
@@ -57,9 +49,21 @@ public void Init()
                     typeof(ImageMagickProvider)));
         }
 
+        private bool IsKnownImageFormat(string path)
+        {
+            try
+            {
+                return new MagickImageInfo(path).Format != MagickFormat.Unknown;
```
Comment (xupefei) [inline QuickLook.Plugin/QuickLook.Plugin.ImageViewer/Plugin.cs:56]: Could you check if an extension is known before calling MagickImageInfo?
Known extensions are given in Line 39--45.

Comment (mooflu) [reply QuickLook.Plugin/QuickLook.Plugin.ImageViewer/Plugin.cs:67]: I've changed it only check extension for well known image and animated image types so it avoids opening the file in those cases.

Comment (mooflu) [reply QuickLook.Plugin/QuickLook.Plugin.ImageViewer/Plugin.cs:67]: Also upgraded Magick.NET to very latest.

**Comment by xupefei** [comment]
Great job, thanks!

**Comment by Hinlavic** [comment]
@mooflu 
Thanks for pulling such a request, and it helps me a lot. However, It seems that the new merged commit has some performance problems.
Once you select one JP2 file, the program will take a long time to open it and not all files with JP2 extension are supported. that is the first problem.
Before the program succeeds to open the first JP2 file, select the next file using arrow keys， it seems that the program failed to release the memory of the fist image. that is the second problem.
image

**Comment by xupefei** [comment]
I disabled detecting images using ImageMagick due to:

- The detection will load the whole file, which is very slow when the file is large,
- Need a better way to handle misdetection. For example, ImageMagick treat all XML files as SVG.

**Pull request merged**